src/App.jsx

```diff
@@ -5,35 +5,43 @@ import NavBar from './components/NavBar/NavBar';
 import Search from './components/Search/Search.jsx';
 import { useSelector } from 'react-redux';
 import Loader from './components/Loader/Loader';
+import { getWeatherInfo } from './redux/slices/weatherSlice';
+import { ThemeProvider } from '@emotion/react';
+import { CssBaseline } from '@mui/material';
+import { getTheme } from './redux/slices/themeSlice';
 
 export const App = () => {
     // const weatherInfo = useWeatherStore((state) => state.weatherInfo);
-    const weatherInfo = useSelector((state) => state.weatherInfoSlice.value);
+    const weatherInfo = useSelector((state) => getWeatherInfo(state));
+    const theme = useSelector((state) => getTheme(state));
 
     return (
         <>
-            <Loader />
-            <NavBar />
-            <Search />
-            {weatherInfo === null ? (
-                <Box
-                    sx={{
-                        display: 'flex',
-                        alignItems: 'center',
-                        justifyContent: 'center',
-                        width: '100%',
-                        height: '75vh',
-                    }}
-                >
-                    <Typography variant='h2'>
-                        Select a city to get started!
-                    </Typography>
-                </Box>
-            ) : (
-                <>
-                    <Dashboard />
-                </>
-            )}
+            <ThemeProvider theme={theme}>
+                <CssBaseline />
+                <Loader />
+                <NavBar />
+                <Search />
+                {weatherInfo === null ? (
```

Comment:
You can create render functions in order to be more readable:
`{weatherInfo === null ? renderCitySelector() : renderDashboard()}`

Reply:
Would those functions just be responsible for returning the relevant components? 
The city selection button are visible at all times for ease of access. (atm they're in the NavBar component)